lib/util.js

```diff
@@ -0,0 +1,56 @@
+const debug = require('./debug')
+const url = require('url')
+const sample = require('lodash.sample')
+
+const getBastion = function (config, baseName) {
+  // If there are bastions, extract a host and a key
+  // otherwise, return an empty Object
+
+  // If there are bastions:
+  // * there should be one *_BASTION_KEY
+  // * pick one host from the comma-separated list in *_BASTIONS
+  // We assert that _BASTIONS and _BASTION_KEY always exist together
+  // If either is falsy, pretend neither exist
+
+  const bastionKey = config[`${baseName}_BASTION_KEY`]
+  const bastionHost = sample((config[`${baseName}_BASTIONS`] || '').split(','))
+  return (!(bastionKey && bastionHost))
+    ? {}
+    : {bastionHost, bastionKey}
+}
+
+exports.getConnectionDetails = function (addon, config) {
+  const connstrings = addon.config_vars
+    .filter((cv) => (
+      config[cv].startsWith('postgres://') &&
+      cv.endsWith('_URL')
+    ))
+
+  if (connstrings.length === 0) {
+    const cli = require('heroku-cli-util')
+    cli.error('Database URL not found for this addon')
+  }
+
+  const baseName = connstrings[0].slice(0, -4)
```

Comment:
That works too. I suggested the regex because it's a simple one and the code is self-documenting, but I'm fine with a comment.